mLRS/CommonRx/mlrs-rx.cpp

```diff
@@ -774,18 +792,55 @@ dbg.puts(s8toBCD_s(stats.last_rssi2));*/
             msp.FrameLost();
         }
 
+#if 0
+stats.received_fhss_index = 63; // enable/disable fhss_index mechanism
+#endif
+
+        // check fhss index
+        // do it only in LISTEN and SYNC
+        if ((connect_state <= CONNECT_STATE_SYNC) && valid_frame_received &&
+            (stats.received_fhss_index < 63)) { // older version don't offer this
+dbg.puts(" check ");
+            if (stats.received_fhss_index_band == 0) {
+                if (fhss.GetCurrI() != stats.received_fhss_index) { // somehow wrong frequency, discard
+                    valid_frame_received = false;
+                    invalid_frame_received = true;
+dbg.puts(" discard");
+                    fhss.SetCurrI(stats.received_fhss_index); // set band 1, should only happen for dual band
+dbg.puts(" set b1");
+                } else { // ok
+                    connect_fhss_index_band_seen |= 0x01; // 0 seen
+                }
+            } else {
+                if (fhss.GetCurrI2() != stats.received_fhss_index) { // somehow wrong frequency, so discard
+                    valid_frame_received = false;
+                    invalid_frame_received = true;
+dbg.puts(" discard");
+                    fhss.SetCurrI2(stats.received_fhss_index); // set band 2, should only happen for dual band
+dbg.puts(" set b2");
+                } else { // ok
+                    connect_fhss_index_band_seen |= 0x02; // 1 seen
+                }
+            }
+        }
+
         if (valid_frame_received) { // valid frame received
             switch (connect_state) {
             case CONNECT_STATE_LISTEN:
                 connect_state = CONNECT_STATE_SYNC;
                 connect_sync_cnt = 0;
+                connect_fhss_index_band_seen = (stats.received_fhss_index < 63) ? 0 : 0x03;
                 break;
             case CONNECT_STATE_SYNC:
                 connect_sync_cnt++;
                 uint8_t connect_sync_cnt_max = CONNECT_SYNC_CNT;
                 if (!connect_occured_once) {
                     connect_sync_cnt_max = Config.connect_sync_cnt_max;
                 }
+                if ((connect_sync_cnt >= connect_sync_cnt_max) && (connect_fhss_index_band_seen != 0x03)) {
+                    connect_sync_cnt = connect_sync_cnt_max - 1; // not yet
+dbg.puts(" delay");
+                }
                 if (connect_sync_cnt >= connect_sync_cnt_max) {
```

Comment:
I would suggest "if (connect_sync_cnt >= connect_sync_cnt_max && connect_fhss_index_band_seen == 0x03) {" and delete the 4 new lines above.